rtmp_to_hls/H264_codec.md

```diff
@@ -0,0 +1,94 @@
+In this chapter you will learn about the H264 codec and how its processing is done with the Membrane Framework plugin - the [H264 plugin](https://github.com/membraneframework/membrane_h264_ffmpeg_plugin).
+
+In H264 we can distinguish two layers of abstraction - the video coding layer (VCL), focused on the visual representation of the video, and the network abstraction layer (NAL), focused on how to structurize the stream so that it can be sent via the network.
+
+When it comes to VCL, a video encoded with the H264 codec is represented as a sequence of pictures. Each picture consists of many **macroblocks**.
+The macroblock is simply a part of the picture (in the context of some spatial dependence) - i.e. the top left corner of the picture.
+The macroblocks are grouped together into so-called **slices**. Each slice consists of some macroblocks, which share the same **slice header**, containing some metadata common for all these macroblocks.
+
+Each slice (which in fact is a part of a video picture), can be packed into a single
+**NALu** (*Network Abstraction Layer units*). As the name suggests, here is where we start to deal with NAL. NALu is just an atomic piece of information sent through the network - it might contain some visual data (i.e. a list of macroblocks forming a part of a video frame), but it is not limited to the visual data - within NALus metadata used to properly decode the stream is also sent. We can distinguish two types of NAL units:
+
+- VCL NALus - which stands for "Video coding layer" NALus
+- Non-VCL NALus - which stands for "Non-video coding layer" NALus
+
+There are different types of both VCL and Non-VCL units - for more information on them you can refer [here](https://yumichan.net/video-processing/video-compression/introduction-to-h264-nal-unit/)
+
+> With the use of Membrane Framework, you can inspect the types of NALus in your h264 file. To do so, you need to clone the H264 parser repository with:
```

Comment:
I think that this blockquote needs a title - what source are we quoting?

Reply:
Hmm, we are using that "tip" notation (with ">") to tag the sections which are not necessary for the user to read (we are not quoting any source).
However, I agree that we need to provide a title

Reply:
done